### PR TITLE
Fix Executor @handler validation for postponed (string) annotations

### PR DESCRIPTION
**Pull request by moonbox3**
### Problem
Using `from __future__ import annotations` stringifies handler annotations. Executor’s `_validate_handler_signature` passed raw string annotations into `validate_workflow_context_annotation`, which relies on `typing.get_origin/get_args` and therefore rejected valid `WorkflowContext[T, U]` annotations.

### Fix
- Evaluate annotations in `_validate_handler_signature` via `inspect.get_annotations(func, eval_str=True)`.
- Use resolved annotations for message_type inference and WorkflowContext validation.

### Tests
- Added `test_executor_future.py` to assert that decorating an `Executor` handler with `ctx: WorkflowContext[MyTypeA, MyTypeB]` under postponed annotations no longer raises and correctly infers `output_types` and `workflow_output_types`.

### CODE REVIEW

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 1]
```diff
@@ -717,25 +717,32 @@ def _validate_handler_signature(
     if len(params) != expected_counts:
         raise ValueError(f"Handler {func.__name__} must have {param_description}. Got {len(params)} parameters.")
 
+    # Resolve annotations (handles `from __future__ import annotations` / string annotations)
+    try:
+        resolved_hints: dict[str, Any] = inspect.get_annotations(func, eval_str=True)
+    except Exception:
+        resolved_hints = {}
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:724]: Avoid broad exception swallowing around `inspect.get_annotations(..., eval_str=True)`: falling back to `{}` can leave ctx/message annotations as strings under postponed evaluation and mask real NameError/ImportError/TypeError. Make the behavior deterministic by raising a clear ValueError (with handler/param context), or narrowly catching expected exceptions, or trying `eval_str=False` as a safer fallback—while never accepting unresolved string annotations as valid.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 2]
```diff
@@ -717,25 +717,32 @@ def _validate_handler_signature(
...
+
     # Check message parameter has type annotation (unless skipped)
     message_param = params[1]
-    if not skip_message_annotation and message_param.annotation == inspect.Parameter.empty:
+    message_annotation = resolved_hints.get(message_param.name, message_param.annotation)
+    if not skip_message_annotation and message_annotation == inspect.Parameter.empty:
         raise ValueError(f"Handler {func.__name__} must have a type annotation for the message parameter")
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:731]: After resolution, explicitly reject/handle string annotations (e.g., unresolved forward refs) for message (and ctx when required). Current logic appears to check presence but not validity, which can allow invalid annotations to slip through and fail later during handler registration/dispatch.

File: python/packages/core/tests/workflow/test_executor_future.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+class MyTypeA:
+    value: int
+
+
+@dataclass
+class MyTypeB:
+    value: str
+
+
+class TestExecutorFutureAnnotations:
```
Comment (moonbox3) [inline python/packages/core/tests/workflow/test_executor_future.py:19]: Add tests for failure/edge paths: (1) unevaluable forward ref should raise a clear, deterministic error (not silently fall back), and (2) decorator-provided types / `skip_message_annotation` path where ctx annotation may be omitted should still work under future annotations.

**Review by moonbox3 — COMMENTED** [review]
## Automated Code Review

**Issue:** #1 | **Reviewers:** 3 | **Confidence:** 72%

### Summary
The change correctly resolves postponed-evaluation annotations before validating WorkflowContext generics and includes a regression test for the reported happy path. However, current error handling around annotation resolution can silently fall back to string annotations, making validation and downstream handler behavior inconsistent and hard to debug. Expand coverage to include failure modes and ensure resolved annotations are deterministically validated or rejected with clear errors.

### Blocking Issues
- [ ] Do not silently swallow exceptions from annotation resolution (e.g., `inspect.get_annotations(..., eval_str=True)`); the current broad `except` + fallback can reintroduce the original postponed-annotations bug (string annotations) and mask genuine NameError/ImportError/TypeError, leading to non-deterministic handler validation/registration. Prefer a deterministic strategy: raise an actionable ValueError, or narrowly catch expected exceptions, or fall back to `eval_str=False` only when safe (and never treat unresolved strings as valid types).
- [ ] Ensure the annotation values used/returned by `_validate_handler_signature()` are consistent with what was validated: today, when resolution fails, `ctx_annotation` can remain a raw string while earlier logic may have validated a different (or no) value, which can cause downstream dispatch/registration behavior to diverge from validation.
- [ ] After attempting resolution, explicitly validate that message (and ctx, when required) annotations are usable types/typing constructs; currently a still-unresolved string annotation may pass presence checks and later break dispatch or produce misleading errors.
- [ ] Add regression tests that cover key edge cases: (1) unevaluable forward refs / missing globals should produce a clear, stable error (not silent fallback), and (2) decorator-provided types / `skip_message_annotation` paths (including omitted ctx annotation) remain supported under future annotations.

### Suggestions
- If you keep a fallback path, consider only falling back when the existing annotations are already non-strings; otherwise fail fast with a targeted error message that names the handler and parameter(s) that could not be evaluated.
- Prefer asserting behavior via public registration APIs in tests when possible; if private fields must be used, keep assertions minimal and focused on externally observable behavior (e.g., registration succeeds and types are as expected).

---
*Automated review by triage-bot*

**Comment by moonbox3** [comment]
## Review Feedback Update

All previously raised blocking issues for issue #1 appear addressed: annotation resolution no longer silently falls back, unresolved string annotations are explicitly rejected, validation now aligns with the resolved values, and regression tests were added for the key edge cases.

Changes pushed in commit `2a83b8202d06eac2ba8d2f429b6875603654938b`.

---
*Automated review by triage-bot*
